jcl/src/jdk.management/share/classes/openj9/lang/management/OpenJ9DiagnosticsMXBean.java

```diff
@@ -71,10 +71,22 @@ public interface OpenJ9DiagnosticsMXBean extends PlatformManagedObject {
 	 * with the initial -Xdump: omitted. See the -Xdump option section on dump agents in
 	 * the documentation for the OpenJ9 JVM.
 	 *
+	 * @return the dump configuration as an array of Strings
 	 * @throws SecurityException if there is a security manager and it doesn't allow the checks required to read the dump settings
 	 */
 	public String[] queryDumpOptions();
 
+	/**
+	 * Returns the current dump configuration as a comma-separated String, or null if an internal error occurs.
+	 * The syntax of the option String is the same as the -Xdump command-line option,
+	 * with the initial -Xdump: omitted. See the -Xdump option section on dump agents in
+	 * the documentation for the OpenJ9 JVM.
+	 *
+	 * @return the dump configuration as a comma-separated String
+	 * @throws SecurityException if there is a security manager and it doesn't allow the checks required to read the dump settings
+	 */
+	public String getDumpOptions();
```

Comment:
Perhaps `TestOpenJ9DiagnosticsMXBean` should be updated to use this new API? As it is, I think `getDumpOptions()` in the test code is broken (it answers the options from a child process instead of the current process); so the check in `resetDumpOptions()` is unaffected by the calls to `setDumpOptions()` or `resetDumpOptions()`.

Reply:
Ok. Looking at this it occurs to me that `,` is not a good separator to use. Do you have a preference for another? I'm thinking `;`, but we could use `;;` as a more complicated separator that is less likely to occur inside an option.
Here are example options, containing two `system:events=` options:
`system:events=gpf+abort+traceassert+corruptcache,file=/home/peter/core.%Y%m%d.%H%M%S.%pid.%seq.dmp,range=1..0,priority=999,request=serial,system:events=systhrow,filter=java/lang/OutOfMemoryError,file=/home/peter/core.%Y%m%d.%H%M%S.%pid.%seq.dmp,range=1..1,priority=999,request=exclusive+compact+prepwalk`

Reply:
Updated the tests.

Reply:
I suggest `|` is preferable to `;` or `;;`.

Reply:
Modified.